Source/SwiftLintFramework/Rules/StatementPositionRule.swift

```diff
@@ -68,17 +135,117 @@ public struct StatementPositionRule: CorrectableRule, ConfigurationProviderRule
         file.write(contents)
         return corrections
     }
+}
+
+//Uncuddled Behaviors
+extension StatementPositionRule {
+    func uncuddledValidateFile(file: File) -> [StyleViolation] {
+        return uncuddledViolationRangesInFile(file).flatMap { range in
+            return StyleViolation(ruleDescription: self.dynamicType.uncuddledDescription,
+                severity: configuration.severity.severity,
+                location: Location(file: file, characterOffset: range.location))
+        }
+    }
 
     // MARK: - Private
 
     // match literal '}'
+    // preceded by whitespace (or nothing)
     // followed by 1) nothing, 2) two+ whitespace/newlines or 3) newlines or tabs
-    // followed by 'else' or 'catch' literals
-    private let pattern = "\\}(?:[\\s\\n\\r]{2,}|[\\n\\t\\r]+)?\\b(else|catch)\\b"
+    // followed by newline and the same amount of whitespace then 'else' or 'catch' literals
+    static let uncuddledPattern = "([ \t]*)\\}(\\n+)?([ \t]*)\\b(else|catch)\\b"
 
-    private func violationRangesInFile(file: File, withPattern pattern: String) -> [NSRange] {
-        return file.matchPattern(pattern).filter { range, syntaxKinds in
-            return syntaxKinds.startsWith([.Keyword])
-        }.flatMap { $0.0 }
+    static let uncuddledRegularExpression = (try? NSRegularExpression(pattern: uncuddledPattern,
+        options: [])) ?? NSRegularExpression()
+
+    static func uncuddledMatchValidator(contents: String) ->
+        ((NSTextCheckingResult) -> NSTextCheckingResult?) {
+        return { match in
+            if match.numberOfRanges != 5 {
+                return match
+            }
+            if match.rangeAtIndex(2).length == 0 {
+                return match
+            }
+            let range1 = match.rangeAtIndex(1)
+            let range2 = match.rangeAtIndex(3)
+            let whitespace1 = contents.substring(range1.location, length: range1.length)
+            let whitespace2 = contents.substring(range2.location, length: range2.length)
+            if whitespace1 == whitespace2 {
+                return nil
+            }
+            return match
+        }
+    }
+
+    static func uncuddledMatchFilter(contents contents: String, syntaxMap: SyntaxMap) ->
+        ((NSTextCheckingResult) -> Bool) {
+        return { match in
+            let range = match.range
+            guard let matchRange = contents.NSRangeToByteRange(start: range.location,
+                                                               length: range.length) else {
+                                                                return false
```

Comment:
could you indent this just one level deeper than the `guard`? I find SourceKit's default here to be absolutely bananas.